### PR TITLE
KAFKA-6053: Fix NoSuchMethodError when creating ProducerRecords with older client versions

### PR DESCRIPTION
**Pull request by apurvam**

### CODE REVIEW

**Comment by apurvam** [comment]
cc @ijuma

**Comment by apurvam** [comment]
I could reproduce the exception locally, and verified that the patch fixes it for the affected tests.

File: tools/src/main/java/org/apache/kafka/tools/VerifiableProducer.java [hunk 1]
```diff
@@ -229,9 +229,15 @@ public static VerifiableProducer createFromArgs(ArgumentParser parser, String[]
 
     /** Produce a message with given key and value. */
     public void send(String key, String value) {
-        ProducerRecord<String, String> record = new ProducerRecord<>(topic, null, createTime, key, value);
-        if (createTime != null)
+        ProducerRecord<String, String> record;
+
+        if (createTime != null) {
```
Comment (ijuma) [inline tools/src/main/java/org/apache/kafka/tools/VerifiableProducer.java]: Can we add a comment explaining why we have to do this?

**Review by ijuma — APPROVED** [review]
LGTM, thanks. Merging to trunk and 1.0 branches.

**Pull request closed without merging**